lib/NetPacksLib.cpp

```diff
@@ -750,9 +746,9 @@ DLL_LINKAGE void NewObject::applyGs(CGameState *gs)
 		o = new CGCreature();
 		{
 			//CStackInstance hlp;
-			CGCreature *cre = static_cast<CGCreature*>(o);
+			auto * cre = dynamic_cast<CGCreature *>(o);

```

Comment:
Please add assert here and in other instances of static_cast -> dynamic_cast change:
```suggestion
			auto * cre = dynamic_cast<CGCreature *>(o);
			assert(cre);
```

Reply:
done